pkg/armhelpers/groupsclient.go

```diff
@@ -18,3 +20,16 @@ func (az *AzureClient) EnsureResourceGroup(name, location string) (resourceGroup
 
 	return &response, nil
 }
+
+// DeleteResourceGroup delete the named resource group
+func (az *AzureClient) DeleteResourceGroup(name string, cancel chan struct{}) error {
+	resCh, errCh := az.groupsClient.Delete(name, cancel)
+	res := <-resCh
+	err := <-errCh
+	// When resourceGroup not exists, we will get 404
```

Comment:
Yes, we should just return the channel and not trying to do too much wrapping and hiding of constructs returned by the clients.